File: DemoProjectKMP-Public/Start/Android/Android_footballmatches/app/src/main/java/com/example/android_footballmatches/domain/usecase/GetLeaguesUseCase.kt
package com.example.android_footballmatches.domain.usecase

import com.example.android_footballmatches.data.model.League
import com.example.android_footballmatches.domain.LeagueRepository


/**
 * Created by Menes SIMEU on 07/06/2024.
 * Volvo MoCE Android Team member
 */
class GetLeaguesUseCase(private val repository: LeagueRepository) {

    suspend operator fun invoke(): List<League> {
        return repository.getLeagues()
    }
}

File: DemoProjectKMP-Public/Start/iOS/iOS_footballmatches/iOS_footballmatches/domain/GetLeaguesUseCase.swift
//
//  GetLeaguesUseCase.swift
//  iOS_footballmatches
//
//  Created by Menes  SIMEU on 07/06/2024.
//
import Foundation

class GetLeaguesUseCase {
    private let repository: LeagueRepository

    init(repository: LeagueRepository) {
        self.repository = repository
    }

    func execute(completion: @escaping (Result<[League], Error>) -> Void) {
        repository.getLeagues(completion: completion)
    }
}
